src/core/Akka/IO/Dns.cs

```diff
@@ -124,11 +124,19 @@ public DnsSettings(Config config)
 
         public DnsExt(ExtendedActorSystem system)
         {
-            _system = system;
-            Settings = new DnsSettings(system.Settings.Config.GetConfig("akka.io.dns"));
-            //TODO: system.dynamicAccess.getClassFor[DnsProvider](Settings.ProviderObjectName).get.newInstance()
-            Provider = (IDnsProvider) Activator.CreateInstance(Type.GetType(Settings.ProviderObjectName));
-            Cache = Provider.Cache;
+            try
+            {
+                _system = system;
+                Settings = new DnsSettings(system.Settings.Config.GetConfig("akka.io.dns"));
+                //TODO: system.dynamicAccess.getClassFor[DnsProvider](Settings.ProviderObjectName).get.newInstance()
+                Provider = (IDnsProvider)Activator.CreateInstance(Type.GetType(Settings.ProviderObjectName, true));
```

Comment:
IMO, `Type.GetType(Settings.ProviderObjectName, true)`should go on a separate line for clarity.
There should also be a check to ensure that the created object from `Activator.CreateInstance` actually is assignable to `IDnsProvder`

The message for the TypeLoadException should indicate that the type is to be used as a Dns Provider.
If the type is not of the correct type, there needs to be a `ConfigurationException` thrown explaining that the type is not a IDnsProvider.